DS3231.cpp

```diff
@@ -35,6 +35,7 @@ Released into the public domain.
 //#include "WProgram.h"
 #include <Arduino.h>
 
+#define UINT32_MAX 4294967295
```

Comment:
Why reinvent the wheel twice, and defining the uint32_MAX separately?
I would use the value of the header <climits>.
See [climits.h](https://cplusplus.com/reference/climits/)